micrometer-core/build.gradle

```diff
@@ -1,3 +1,7 @@
+plugins {
+    id "org.jetbrains.kotlin.jvm" version "1.7.0"
```

Comment:
After this pull request, something is adding the kotlin standard library as a non-optional dependency to micrometer-core. I suspect it is this plugin.
https://kotlinlang.org/docs/gradle.html#dependency-on-the-standard-library